Standardize example descriptions/comments

softwareengineerprogrammer opened the pull request:
You can see the new descriptions at https://gtp.scientificwebservices.com/geophires -> Examples tab

<img width="1240" alt="Screenshot 2024-06-19 at 06 30 13" src="https://github.com/NREL/GEOPHIRES-X/assets/4056124/01e3b3b7-a3e3-4361-9c8a-b0ebdf5cb2dc">

malcolm-dsider reviewed (changes requested):
This looks fine. It seems the new rule is that comments start with "#". Can we update the input parameter line parsing code to check for lines that start with "#" and automatically ignore them, no matter what they contain? I ask for this because what I have noted in the past that if the comment lines contains commas, the parser will try to parse the line and will fail, producing an error. I went thru and removed all the commas in comments, but now if we use "#" to designate comments, we can have as many commas in the comments as we want if the parser code is updated to handle the new comment line delimiter. I suggest we try to maintain backward compatibility for comments that don't start with "#" as much as possible, with the caveat that those lines must not contain commas.

softwareengineerprogrammer commented:
> This looks fine. It seems the new rule is that comments start with "#". Can we update the input parameter line parsing code to check for lines that start with "#" and automatically ignore them, no matter what they contain? I ask for this because what I have noted in the past that if the comment lines contains commas, the parser will try to parse the line and will fail, producing an error. I went thru and removed all the commas in comments, but now if we use "#" to designate comments, we can have as many commas in the comments as we want if the parser code is updated to handle the new comment line delimiter. I suggest we try to maintain backward compatibility for comments that don't start with "#" as much as possible, with the caveat that those lines must not contain commas.

Good call - Added test cases with a variety of comment styles, and discovered and fixed some edge cases in doing so.
